content/docs/iac/using-pulumi/extending-pulumi/build-a-component.md

```diff
@@ -0,0 +1,1808 @@
+
+{{% choosable language csharp %}}
+
+***Example:** A `Pulumi.yaml` file for a Pulumi project written in C#*
+
+```yaml
+name: use-static-page-component
+description: A minimal C# Pulumi program that uses the custom Static Page component
+runtime:
+  name: dotnet
+```
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+***Example:** A `Pulumi.yaml` file for a Pulumi project written in Java*
+
+```yaml
+name: use-static-page-component
+description: A minimal Java Pulumi program that uses the custom Static Page component
+runtime:
+  name: java
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+***Example:** A `Pulumi.yaml` file for a Pulumi project written in YAML*
+
+```yaml
+name: use-static-page-component
+description: A minimal YAML Pulumi program that uses the custom Static Page component
+runtime:
+  name: yaml
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+#### Generate the SDK
+
+In order to use our Pulumi component from source, we will need to generate a language-specific SDK, which will allow end users to use it in a Pulumi program. From the root directory of your `use-static-page-component` Pulumi project, run the following command:
+
+```bash
+pulumi package add ../static-page-component
+```
+
+This will create a new subdirectory called `sdks/`, and in it, there will be a directory for your language SDK. One of: `dotnet`, `go`, `java`, `nodejs`, and `python`.
+
+{{% notes type="tip" %}}
+If you're authoring your Pulumi project in YAML, it is not necessary to generate a SDK! Skip ahead to the next step.
+{{% /notes %}}
+
+#### Add the component reference in Pulumi.YAML
+
+Now, in your `Pulumi.yaml` file add the following section to load the component from source:
+
+```yaml
+plugins:
+  providers:
+    - name: static-page-component
+      path: ../static-page-component
+```
+
+{{< chooser language "javascript,typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language javascript %}}
+
+#### Add the NodeJS project file
+
+Now lets create our `package.json`. We'll need the standard `pulumi` SDK and our custom component. To use that, just add the path to the generated JavaScript SDK using the format `file:<path>` instead of a package version spec.
+
+***Example:** `package.json`*
+
+```json
+{
+    "name": "use-static-page-component",
+    "dependencies": {
+        "@pulumi/pulumi": "3.157.0",
+        "@pulumi/static-page-component": "file:sdks/static-page-component"
+    }
+}
+```
+
+Note that we don't need to add the Pulumi AWS provider library here, because that dependency is handled by the component project, in whatever langauge you implemented it in. We just need to carry a reference to the component SDK which provides us access to the component via RPC to its provider host. This creates a clean separation of concerns between the component implmentation and the end users of the component.
+
+#### Install dependencies
+
+Next, install the `pulumi` dependencies:
+
+```bash
+pulumi install
```

Comment:
If you use the `packages` section, `pulumi install` will create SDKs for you, so you don't need to run `pulumi package add`.

Reply:
good to know!